packages/uicore/src/components/ModalDialog/ModalDialog.tsx

```diff
@@ -83,7 +84,7 @@ export const ModalDialog: FC<ModalDialogProps> = ({
   const initScrollShadows = useCallback(() => {
     if (!bodyRef.current) return
 
-    bodyObserverRef.current?.disconnect()
+    isFunction(bodyObserverRef.current?.disconnect) && bodyObserverRef.current?.disconnect()
```

Comment:
```suggestion
     bodyObserverRef.current?.disconnect?.()
```

Reply:
@PratyushGar This null check will not check field is executable or not and still error will be there.